website/src/pages/login.page.tsx

```diff
@@ -0,0 +1,176 @@
+import { CognitoUser } from "amazon-cognito-identity-js"
+import React from "react"
+import {
+  unstable_Form as Form,
+  unstable_FormInput as FormInput,
+  unstable_FormLabel as FormLabel,
+  unstable_FormMessage as FormMessage,
+  unstable_FormSubmitButton as FormSubmitButton,
+  unstable_FormStateReturn,
+  unstable_useFormState as useFormState,
+} from "reakit/Form"
+import { Popover, PopoverDisclosure, usePopoverState } from "reakit/Popover"
+import { useUser } from "src/auth"
+import Link from "src/components/link"
+import { centeredColumn } from "src/sprinkles.css"
+import Layout from "../layout"
+import {
+  centeredForm,
+  loginFormBox,
+  loginHeader,
+  logoutPopover,
+  positionButton,
+  skinnyWidth,
+  submitButton,
+} from "./login.css"
+import { ResetLink } from "./reset-password.page"
+
+interface FormFieldsType {
+  form: unstable_FormStateReturn<any | undefined>
+  name: any
+  label: string
+  type?: string | undefined
+  placeholder: string
+}
+
+export const FormFields = ({
+  form,
+  name,
+  label,
+  type,
+  placeholder,
+}: FormFieldsType) => {
+  return (
+    <>
+      <FormLabel {...form} name={name} label={label} />
+
+      <FormInput
+        {...form}
+        name={name}
+        className={loginFormBox}
+        type={type}
+        placeholder={placeholder}
+      />
+    </>
+  )
+}
+
+const LoginPage = () => {
+  const { loginUser } = useUser().operations
+
+  const loginForm = useFormState({
+    values: { username: "", password: "" },
+    onValidate: (values) => {
+      if (!values.username || !values.password) {
+        throw (
+          (!values.username && { username: "A username is required" }) || {
+            password: "A password is required",
+          }
+        )
+      }
```

Comment:
I think this could just be

```ts
if (!values.username) {
  throw ...
} else if (!values.password) {
  throw ...
}
```

Or perhaps something like this, to show both messages if both are missing:

```ts
if (!values.username || !values.password) {
  throw { username: !values.username && "...", password: !values.password && "..." }
}
``

Reply:
Modified!